### PR TITLE
Feature/walls

### PR DESCRIPTION
**Pull request by cch1**
Add random walls to arena, ensuring that no partitioning of arena occurs.

TODO: update visiible arena functionality

### CODE REVIEW

File: src/clj/battlebots/arena/generation.clj [hunk 1]
```diff
@@ -37,31 +38,86 @@
   [amount item arena]
   (reduce replacer arena (repeat amount item)))
 
+(defn- wrap
```
Comment (oconn) [inline src/clj/battlebots/arena/generation.clj]: @cch1 `wrap` & `move` are very well written. I have two functions in `arena.utils`, `wrap-coords` & `adjust-coords` that accomplish a similar goal, but I'd like to replace them with yours. The signatures are similar, the biggest difference I can see is that `adjust-coords` allows you to apply a distance to adjust, however always wraps. `move` appears to always adjust one space, but allows for an oob coord to be calculated (defaults to wrap)? I wired the tests I wrote for `wrap-coords` up to `wrap` and they all pass without modification, so I believe we can just drop yours in it's place. What's your thoughts on `move` & `adjust-coords`?

File: src/clj/battlebots/arena/generation.clj [hunk 2]
```diff
@@ -37,31 +38,86 @@
...
+  "Wraps out-of-bounds coordinates (zero-based) to opposite edge of m x n arena"
+  [[x y] [m n]]
+  {:pre [(integer? x) (integer? y) (pos? m) (pos? n)]
```
Comment (oconn) [inline src/clj/battlebots/arena/generation.clj]: `pre` and `post` assertion functions are cool! I had no idea you could do something like this.

Comment (cch1) [inline src/clj/battlebots/arena/generation.clj]: Yep.  And they have access to destructured args *and* can be set per-signature for polymorphic (by arg count) functions.

File: src/clj/battlebots/arena/generation.clj (hunk 1 above)
Comment (cch1) [inline src/clj/battlebots/arena/generation.clj]: Sounds great to me.

**Comment by oconn** [comment]
Added https://github.com/willowtreeapps/battlebots/issues/66 to cover the above conversation.

**Pull request merged**